100-level-AsyncAPI-Intro/step1.md

```diff
@@ -0,0 +1,54 @@
+
+Eve begins by providing Chan with a comprehensive explanation of event-driven architecture (EDA) and outlines the process of designing an EDA system using AsyncAPI.
+
+## Event-driven architecture
+
+An event signifies a change in the system's state that triggers communication between different services. Event-driven architecture (EDA) decomposes an application into several components interacting via events. These components encompass event producers and consumers:
+
+- An event **producer** identifies an event and transmits relevant information through the event channel.
+- An event **consumer** subscribes to an event channel to receive and process event data.
+
+Each component can fulfill the role of an event producer, consumer, or both.
+
+As Eve continues her conversation with Chan, she realizes she hasn't offered him a beverage. She quickly rushes to prepare a cup of coffee. While making the coffee, she draws upon a coffee shop as an analogy: a customer (producer) places an order for an espresso (event), conveying the order details to the barista (consumer). The sequence unfolds as follows:
+
+1. The customer initiates an espresso order (an event), and the barista receives the order details.
+2. Subsequently, the barista processes the order and begins crafting the espresso (a response to the event).
+
+In this coffee shop scenario, event-driven architecture mirrors the seamless flow of orders and drinks without any disruption, ensuring a smooth and efficient operation.
+Chan asks Eve how the AsyncAPI specification impacts the event-driven architecture world.
+
+## Event-driven architecture with AsyncAPI
+
+AsyncAPI simplifies the definition of communication and message exchange among different components in an Event-Driven Architecture (EDA). It provides a structured way to describe events, channels, and message formats, enhancing overall understanding.
+
+AsyncAPI helps you define asynchronous APIs. Asynchronous APIs are like digital mailboxes where you can send messages without waiting for an instant reply. When the other person reads your letter and has something to say, they will reply. AsyncAPI maps out events and their responses in an EDA.
+
+Chan spots cake in Eve's kitchen and asks for a slice as she likes cakes. While offering her a slice, Eve gives an example: imagine a birthday party with events like cake cutting and gift opening. These events trigger responses, such as singing during the cake-cutting or clapping while opening the gifts.
+
+AsyncAPI helps organize events, determining when they'll occur and how everyone should react. For example, during the cake cutting, AsyncAPI guides in singing the birthday song. It also enables the creation of a party schedule, like cutting the cake at 6 pm and opening gifts at 7 pm.
+
+Here are reasons why using AsyncAPI enhances EDA design:
+
+* AsyncAPI plans and describes each event and its corresponding reactions across different components.
```

Comment:
I don't fully agree on this one. Related with the previous sentence `determining when they'll occur and how everyone should react`.

AsyncAPI doesn't describe the message flow per se, except in the case of a request/reply, where it just defines one expected interaction but nothing else. For describing the flows, https://serverlessworkflow.io is the closest specification.

Reply:
Understood.